In [1]:
import math
import pandas as pd
import numpy as np
from scipy.constants import Avogadro
from scipy.constants import hbar
from scipy.constants import c

# **12.4(a)**

In [31]:
# Declaración de variables
uma_h = 1.008
uma_c = 12.011
uma_cl = 34.9688
r_norm = 178*(10**-12)
r_prim = 111*(10**-12)
angle_rad_12_4 = math.radians(111)
#Conversión de unidades
masa_h = uma_h / (1000*Avogadro)
masa_c = uma_c / (1000*Avogadro)
masa_cl = uma_cl / (1000*Avogadro)
masa_total_12_4 = masa_h + masa_c + masa_cl
print(f"Masa total = {masa_total_12_4:.3e} kg")

Masa total = 7.969e-26 kg


Expresiones para el momento:
$$I_{∥} = 2m_{Cl} (1-\cos{\theta})R^{2}$$

$$I_{\perp} = m_{Cl}(1-\cos{\theta})R^{2} + \frac{m_{Cl}}{M}(m_{C}+m_{H})(1+2\cos{\theta})R^{2}+ \frac{m_H}{M} \{ (3m_{Cl} + m_{C})R^{'} + 6 m_{Cl}R [ \frac{1}{3} (1+2 \cos{\theta})]^{1/2} \} R^{'}$$

$$ A = \frac{\hbar}{4\pi c I_{\parallel}}$$

$$ B = \frac{\hbar}{4\pi c I_{\perp}}$$

In [32]:
#Momento de inercia
I_paralel = 2*masa_cl*(1-math.cos(angle_rad_12_4))*(r_norm**2)
a_perp = masa_cl*(1-math.cos(angle_rad_12_4))*(r_norm**2)
b_perp = (masa_cl/masa_total_12_4)*(masa_c+masa_h)*(1+2*math.cos(angle_rad_12_4))*(r_norm**2)
c_perp = (masa_h/masa_total_12_4) * ((
    (3*masa_cl + masa_c) * r_prim)+
 6*masa_cl*r_norm*((1 + 2*math.cos(angle_rad_12_4))/3)**(0.5))*(r_prim**2)
I_perp = a_perp + b_perp + c_perp

In [33]:
print(a_perp)
print(b_perp)
print(c_perp)

2.49912084723378e-45
1.41386569283281e-46
1.050920969158922e-56


In [5]:
print(f"I (Primario) = {I_paralel:.3e}")
print(f"I (Secundario) = {I_perp:.3e}")

I (Primario) = 4.998e-45
I (Secundario) = 2.641e-45


Expresiones constantes rotacionales
$$ \widetilde{A}  = \frac{\hbar}{4\pi c I_{∥}}$$

$$ \widetilde{B} = \frac{\hbar}{4\pi c I_{\perp}}$$

In [28]:
cons_a_virgul = hbar/(4*math.pi*c*I_paralel)
cons_b_virgul = hbar/(4*math.pi*c*I_perp)
cons_a = hbar/(4*math.pi*I_paralel)
cons_b = hbar/(4*math.pi*I_perp)

In [29]:
print("Eje Primario =", I_paralel, "Eje Secundario =", I_perp)
print(f"A virgul ={cons_a_virgul:.2f} cm^{-1}")
print(f"B virgul ={cons_b_virgul:.2f} cm^{-1}")
print(f"A = {cons_a:.3e} Hz")
print(f"B = {cons_b:.3e} Hz")



Eje Primario = 4.99824169446756e-45 Eje Secundario = 2.64050741652757e-45
A virgul =5.60 cm^-1
B virgul =10.60 cm^-1
A = 1.679e+09 Hz
B = 3.178e+09 Hz


# **12.5(a)**
Calculate the frequency of the J = 4←3 transition in the pure
rotational spectrum of $^{14}\text{N}^{16}\text{O}$. The equilibrium bond length is 115 pm.

$$\widetilde{F}(J) = \widetilde{B}J(J+1)$$

$$I = \mu R^2 \; \; \text{con} \; \; \mu = \frac{m_{A}m_{B}}{M}$$
Sea $m(^{14}N) = 14.007 \: \text{uma}$ \& $m(^{16}O) = 15.999 \: \text{uma}$

In [8]:
uma_n = 14.007
uma_o = 15.999
masa_n = uma_n / (1000*Avogadro)
masa_o = uma_o / (1000*Avogadro)
masa_total_12_5 = masa_n + masa_o
masa_reducida_12_5 = (masa_n*masa_o) / masa_total_12_5
r_12_5 = 115*(10**-12)

In [19]:
#Momento de inercia
I_12_5 = masa_reducida_12_5*(r_12_5**2)
#Constante rotacional
const_b_12_5 = (hbar)/(4*math.pi* I_12_5)
#Frecuencia
nu_12_5 = 2*const_b_12_5*(3+1)

In [20]:
print(f"B = {const_b_12_5:.3e} Hz")
print(f"v = {nu_12_5:.3e} Hz")

B = 5.117e+10 Hz
v = 4.093e+11 Hz


# **12.6(a)**
If the wavenumber of the J = 3←2 rotational transition of $^1\text{H}^{35}\text{Cl}$
considered as a rigid rotator is 63.56 cm−1, what is (a) the moment of inertia
of the molecule, (b) the bond length?

Sea $\widetilde{\nu} = 63.56 \; \text{cm}^{-1} $


$63.56 \; \text{cm}^{-1} = 2\widetilde{B}(J+1)$

$\widetilde{B} = \frac{\hbar}{4 \pi I c} \rightarrow I = \frac{\hbar}{ 4 c \pi \widetilde{B}}$

$ I = \mu R^{2} $

In [49]:
# Momento de inercia
const_b_12_6 = 63.56/6
I_12_6 = (hbar)/(4*math.pi*const_b_12_6*(100)*c)
print(f"I = {I_12_6:.3e} kg m^2")

I = 2.642e-47 kg m^2


In [54]:
# Longitud del enlace
masa_reducida_12_6 = masa_h*masa_cl / (masa_h+masa_cl)
longitud_12_6 = ((I_12_6/masa_reducida_12_6)**(1/2))/1*10**12
print(f"R = {longitud_12_6:.3f} pm")

R = 127.445 pm


# **12.7(a)**
Given that the spacing of lines in the microwave spectrum of $^{27}\text{Al}^1\text{H}$ is
constant at 12.604 cm$^{−1}$, calculate the moment of inertia and bond length of
the molecule. (m($^{27}$Al) = 26.9815 mu.)

$$\Delta \widetilde{\nu} = 12.604 \; \text{cm}^{-1} = 2\widetilde{B}$$

In [63]:
#Momento de inercia
const_b_12_7 = (12.604)/2
I_12_7 = hbar / (4*math.pi*const_b_12_7*(100)*c)
print(f"I = {I_12_7:.3e} kg m^2")
#Longitud del enlace
masa_al = 26.9815 / (1000*Avogadro)
masa_reducida_12_7 = masa_al*masa_h / (masa_al+masa_h)
longitud_12_7 = ((I_12_7/masa_reducida_12_7)**(1/2))/1*10**12
print(f"R = {longitud_12_7:.3f} pm")

I = 4.442e-47 kg m^2
R = 165.918 pm


# **12.8(a)**
The rotational constant of $^{127}\text{I}^{35}\text{Cl}$ is 0.1142 cm$^{−1}$. Calculate the ICl
bond length. (m($^{35}$Cl) = 34.9688mu, m($^{127}$I) = 126.9045mu.)

$$\widetilde{B} = 0.1142 \; \text{cm}^{-1}$$

Para rotores lineales triatómicos se tiene:

$$I = \mu R ^{2}$$

Donde A, corresponde a los homoátomos que se encuentran en los extremos de cada enlace.


In [79]:
#Momento de inercia
const_b_12_8 = 0.1142
I_12_8 = hbar / (4*math.pi*const_b_12_8*(100)*c)
print(f"I = {I_12_8:.3e} kg m^2")

#Longitud de enlace
masa_I = 126.9045 / (1000*Avogadro)
masa_reducida_12_8 = masa_I*masa_cl / (masa_I+masa_cl)
longitud_12_8 = ((I_12_8/(masa_reducida_12_8))**(1/2))/ 1*10**12
print(f"R = {longitud_12_8:.3f} pm")

I = 2.451e-45 kg m^2
R = 232.046 pm


# **12.9(a)**
Determine the HC and CN bond lengths in HCN from the rotational
constants B($^{1}\text{H}^{12}\text{C}^{14}\text{N}$) = 44.316 GHz and B($^{2}\text{H}^{12}\text{C}^{14}\text{N}$) = 36.208 GHz.

In [88]:
const_b_12_9_1 = 44.316 * 10**(9)
const_b_12_9_2 = 36.208 * 10**(9)
#Momentos de inercia
I_12_9_1 = hbar / (4*math.pi*const_b_12_9_1)
I_12_9_2 = hbar / (4*math.pi*const_b_12_9_2)
print(f"I_1 = {I_12_9_1:.3e} kg m^2")
print(f"I_2 = {I_12_9_2:.3e} kg m^2")


I_1 = 1.894e-46 kg m^2
I_2 = 2.318e-46 kg m^2
